components/datetime/src/skeleton/helpers.rs

```diff
@@ -317,6 +319,34 @@ fn adjust_pattern_field_lengths(fields: &[Field], pattern: &mut Pattern) {
     })
 }
 
+/// Alters given Pattern so that it will have a fractional second field if it was requested.
+///
+/// If the requested skeleton included both seconds and fractional seconds and the dateFormatItem
+/// skeleton included seconds but not fractional seconds, then the seconds field of the corresponding
+/// pattern should be adjusted by appending the locale’s decimal separator, followed by the sequence
+/// of ‘S’ characters from the requested skeleton.
+/// (see https://unicode.org/reports/tr35/tr35-dates.html#Matching_Skeletons)
+fn append_fractional_seconds(pattern: &mut Pattern, fields: &[Field]) {
```

Comment:
Thought (this is an existing issue, so you are in no obligation to do any action on it, I just wanted to bring it up): 

Having `icu_datetime::pattern::reference::pattern::Pattern` and `icu_datetime::pattern::runtime::pattern::Pattern` makes this super confusing to review, as opposed to having a struct named `ReferencePattern` and `RuntimePattern`. `Pattern` is pretty much impossible to grep in the codebase. There's even another `Pattern` struct at `utils/pattern/src/pattern/mod.rs`.

In addition, since these aren't publicly exposed, there's not documentation on the structs themselves, which leads to the awkward situation in the editor where no documentation is provided:

<img width="486" alt="image" src="https://user-images.githubusercontent.com/1588648/167944509-b975a2ff-04c1-4e37-8f4e-d093392afa41.png">

This is mostly a bad code reading experience, which makes it hard to reason with some code changes like the ones in this function that are moving between reference and runtime patterns.

Reply:
Filed https://github.com/unicode-org/icu4x/issues/1889 for this.